cmd/cmd.go

```diff
@@ -937,6 +1032,53 @@ func RunServer(cmd *cobra.Command, _ []string) error {
 	return server.Serve(ln, origins)
 }
 
+func getImageData(filePath string) ([]byte, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	buf := make([]byte, 512)
```

Comment:
bytes.Buffer is probably more appropriate. you can do something like this where it just appends to the 512

```go
var b bytes.Buffer

if _, err := io.CopyN(&b, file); err != nil {
  // return err
}

contentType := http.DetectContentType(b.Bytes())
if !slices.Contains(types, contentType) {
  // return err
}

if _, err := io.Copy(&b, file); err != nil {
  // return err
}
```

Reply:
We still have to stat the file and only read if the file is < 100MB. I feel like we're splitting hairs here.